aptos-move/aptos-vm/src/aptos_vm_impl.rs

```diff
@@ -215,13 +217,17 @@ impl AptosVMImpl {
             let valid = if let Ok(Some(data)) = data {
                 let approved_execution_hashes =
                     bcs::from_bytes::<ApprovedExecutionHashes>(&data).ok();
-                approved_execution_hashes
+                let valid = approved_execution_hashes
                     .map(|aeh| {
                         aeh.entries
                             .into_iter()
                             .any(|(_, hash)| hash == txn_data.script_hash)
                     })
-                    .unwrap_or(false)
+                    .unwrap_or(false);
+                valid
+                    && (txn_data.script_size + txn_gas_params.max_transaction_size_in_bytes
```

Comment:
what does the second condition mean? data size < 64KB?

Reply:
yes